backend/stream/crawlers/news_crawler.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import json
+import os
+import sys
+from datetime import datetime
+
+import yaml
+
+from backend.stream.apis.diffbot import DiffbotArticleClient
+from backend.stream.apis.news_api import NewsApiClient
+from backend.stream.utils.db_util import update_article_url_db
+from backend.stream.utils.file_util import make_parent_dirs
+from backend.stream.utils.misc_util import overwrite_config
+
+
+def get_argparser():
+    parser = argparse.ArgumentParser(description='News crawler')
+    parser.add_argument('--config', required=True, help='config file path')
+    parser.add_argument('--json', help='dictionary to overwrite config')
+    parser.add_argument('--tol', type=int, default=0, help='maximum number of News API errors you can tolerate')
+    parser.add_argument('--db', required=True, help='output DB file path')
+    parser.add_argument('--output', required=True, help='output root dir path')
+    return parser
+
+
+def get_related_article_urls(news_api_config, max_tol, timestamp, category, db_file_path):
+    article_dict_list = list()
+    news_api_client = NewsApiClient()
```

Comment:
Try to decouple the `ApiClient` from this function call (for instance you could instatiate the client outside the function and pass  it as an argument). This way you can create a mock client (that always returns the same response) so that you can test this function without the external dependency on the API.

Reply:
Sure, I just decoupled it from the function